sdk/src/utils/errorHandler.ts

```diff
@@ -5,12 +5,15 @@ export function handleError(type: ActionType, err: unknown): never {
   if (err instanceof BaseError) {
     const revertError = err.walk((err) => err instanceof ContractFunctionRevertedError);
     if (revertError instanceof ContractFunctionRevertedError) {
-      const errorName = revertError.data?.errorName ?? "";
+      const errorName = revertError.data?.errorName ?? revertError.signature ?? "unknown revert reason";
```

Comment:
Let's avoid this nested ternary expression, it's quite hard to read/maintain.
Instead you could create a small `extractErrorName(revertError: ContractFunctionRevertedError)` function to make it easier to change the wording or content when throwing our error.

Also, in the default case, `errorName` should be "an unknown revert error" to make the full message perfect in the end.